stdlib/asyncio/streams.pyi

```diff
@@ -160,9 +160,9 @@ class StreamReader(AsyncIterator[bytes]):
     def set_transport(self, transport: transports.BaseTransport) -> None: ...
     def feed_eof(self) -> None: ...
     def at_eof(self) -> bool: ...
-    def feed_data(self, data: bytes) -> None: ...
+    def feed_data(self, data: Iterable[SupportsIndex]) -> None: ...
     async def readline(self) -> bytes: ...
-    async def readuntil(self, separator: bytes = ...) -> bytes: ...
+    async def readuntil(self, separator: ReadableBuffer = ...) -> bytes: ...
```

Comment:
This code is using `len(separator)`. Is it guaranteed that all buffer protocol objects support `len()`? While buffer objects have fields like `len` and `itemsize`, they refer to the in-memory presentation as far as I can see.

Unfortunately, https://docs.python.org/3/c-api/buffer.html only describes the C side of things, but not how a buffer protocol object looks on the Python side, so it's a bit hard for me to judges buffer uses on the Python-side.

Reply:
Good catch, the buffer protocol does not require `len()`. My favorite bare-bones buffer, `pickle.PickleBuffer`, has no len(). I'll just change it to `bytes | bytearray | memoryview` with a comment; after PEP 688 is available we could use a protocol with `__buffer__` and `__len__`.